### DISCUSSION TITLE
如何使用 LMDeploy 把 InternLM-20B-4bit 部署为服务

### DISCUSSION THREAD

**Discussion by vansin** [InternLM/lmdeploy#451] [Show and tell]
通过以下步骤，即可快速把 InternLM-20B-4bit 部署为服务，并在线与模型聊天。

第一步：安装 lmdeploy

```shell
pip install 'lmdeploy>=0.0.9'
```
第二步：下载 InternLM-20B-4bit 模型

```python
git-lfs install
git clone --depth 1 https://huggingface.co/internlm/internlm-chat-20b-4bit
```

第三步：转换模型权重格式

```python
python3 -m lmdeploy.serve.turbomind.deploy internlm-chat \
    --model-path ./internlm-chat-20b-4bit \
    --model-format awq \
    --group-size 128
```
第四步：启动 gradio 服务

```shell
python3 -m lmdeploy.serve.gradio.app ./workspace --server_name {ip_addr} --server_port {port}
```

**Comment by vansin** [comment]
通过以下步骤，即可快速把 InternLM-20B-Chat 部署为服务，并在线与模型聊天。

第一步：安装 lmdeploy

```shell
pip install 'lmdeploy>=0.0.9'`
```
第二步：下载 InternLM-20B-4bit 模型

```python
git-lfs install
git clone https://huggingface.co/internlm/internlm-chat-20b
```

第三步：转换模型权重格式

```python
python3 -m lmdeploy.serve.turbomind.deploy internlm-chat \
    --model-path ./internlm-chat-20b
```
第四步：启动 gradio 服务

```shell
python3 -m lmdeploy.serve.gradio.app ./workspace --server_name {ip_addr} --server_port {port}
```

**Comment by YIBO0408** [comment]
cd internlm-chat-20b
git lfs pull

**Comment by luoqingyi1993** [comment]
![81a1b0de0c36fa4e37011aea03bddf7](https://github.com/InternLM/lmdeploy/assets/50549044/4fe33897-638c-47df-a378-bc2034c96261)

量化部署出现这个错误，大佬们看看是为什么？

**Reply by lvhan028** [reply]
如字面意思。`ll ./root/model/internlm-chat-20b-4bit/` 的结果贴一下吧

**Reply by luoqingyi1993** [reply]
这个是文件夹，里面放的就是4bit的模型

**Reply by lvhan028** [reply]
日志显示：./root/model/internlm-chat-20b-4bit/tokenizer.model doesn't exist.
文件夹内的文件是全的不？

**Reply by luoqingyi1993** [reply]
全的呢

**Reply by lvhan028** [reply]
我复现不出来。能贴一下 `ll ./root/model/internlm-chat-20b-4bit/ ` 的结果吗？

**Reply by luoqingyi1993** [reply]
![image](https://github.com/InternLM/lmdeploy/assets/50549044/1f6e1372-6475-4538-aa19-fe835fb636f9)

**Comment by luoqingyi1993** [comment]
24G的显存，好像跑不了4bit的量化模型？还是哪里的参数要设置？

**Reply by lvhan028** [reply]
triton_models/weights/config.ini中的session_len调小
